src/ui/public/query_bar/components/query_bar.tsx

```diff
@@ -0,0 +1,490 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IndexPattern } from 'ui/index_patterns';
+
+declare module '@elastic/eui' {
+  export const EuiOutsideClickDetector: SFC<any>;
+}
+
+import { debounce } from 'lodash';
+import React, { Component, SFC } from 'react';
+import { getFromLegacyIndexPattern } from 'ui/index_patterns/static_utils';
+import { PersistedLog } from 'ui/persisted_log';
+import { Storage } from 'ui/storage';
+import {
+  AutocompleteSuggestion,
+  AutocompleteSuggestionType,
+  getAutocompleteProvider,
+} from '../../autocomplete_providers';
+import chrome from '../../chrome';
+import { fromUser, toUser } from '../../parse_query';
+import { matchPairs } from '../lib/match_pairs';
+import { QueryLanguageSwitcher } from './language_switcher';
+import { SuggestionsComponent } from './typeahead/suggestions_component';
+
+import { EuiFieldText, EuiOutsideClickDetector } from '@elastic/eui';
+
+const KEY_CODES = {
```

Comment:
Pretty sure these can be imported from EUI: `import { keyCodes } from '@elastic/eui';`

Reply:
Just checked, unfortunately EUI doesn't include `home` or `end`